allensdk/brain_observatory/ecephys/utils.py

```diff
@@ -65,3 +66,57 @@ def clobbering_merge(to_df, from_df, **kwargs):
 
     to_df = to_df.drop(columns=list(overlapping))
     return pd.merge(to_df, from_df, **kwargs)
+
+
+def strip_substructure_acronym(
+        acronym: Optional[Union[str, list, float]]
+) -> Optional[Union[str, list]]:
+    """
+    Sanitize a structure acronym or a list of structure acronyms
+    by removing the substructure (e.g. DG-mo becomes DG).
+
+    If acronym is a list, every element in the list will be sanitized
+    and a list of unique acronyms will be returned. **Element order will
+    not be preserved**.
+
+    If acronym is None, return None. If None occurs in a list of
+    sturture acronyms, it will be omitted
+
+    Note: if acronym is NaN, it will get converted to None;
+    any other float will provoke an error.
+    """
+
+    if isinstance(acronym, numbers.Number):
+        if np.isnan(acronym):
+            acronym = None
+
+    if isinstance(acronym, str):
+        return acronym.split('-')[0]
+    elif isinstance(acronym, list):
+        new_acronym = set()
+
+        for el in acronym:
+            if isinstance(el, str):
```

Comment:
It looks like this part can recursively call `strip_substructure_acronym` and add the results to a list if you feel comfortable writing that

Reply:
Good catch; done.